### PR TITLE
Bug 1279555 - Fix cursor sharing, port required code from old conversation.jsx to sidebar.jsx.

### PR DESCRIPTION
**Pull request by Standard8**

### CODE REVIEW

File: add-on/panels/js/sidebar.jsx [hunk 1]
```diff
@@ -12,12 +12,43 @@ loop.sidebar = (function(mozL10n) {
...
+    componentWillMount: function() {
+      this.listenTo(this.props.cursorStore, "change:remoteCursorPosition",
+                    this._onRemoteCursorPositionChange);
+      this.listenTo(this.props.cursorStore, "change:remoteCursorClick",
+                    this._onRemoteCursorClick);
+    },
+
+    _onRemoteCursorPositionChange: function() {
+      loop.request("AddRemoteCursorOverlay",
+                  this.props.cursorStore.getStoreState("remoteCursorPosition"));
```
Comment (Mardak) [inline add-on/panels/js/sidebar.jsx]: nit: 1 more space

File: add-on/panels/js/sidebar.jsx [hunk 2]
```diff
@@ -12,12 +12,43 @@ loop.sidebar = (function(mozL10n) {
...
+    },
+
+    _onRemoteCursorClick: function() {
+      let click = this.props.cursorStore.getStoreState("remoteCursorClick");
+      // if the click is 'false', assume it is a storeState reset,
+      // so don't do anything
```
Comment (Mardak) [inline add-on/panels/js/sidebar.jsx]: nit: comment casing/punctuation

File: add-on/panels/js/sidebar.jsx [hunk 3]
```diff
@@ -158,23 +189,28 @@ loop.sidebar = (function(mozL10n) {
       var textChatStore = new loop.store.TextChatStore(dispatcher, {
         dataDriver: dataDriver
       });
+      var remoteCursorStore = new loop.store.RemoteCursorStore(dispatcher, {
+        sdkDriver: sdkDriver
```
Comment (Mardak) [inline add-on/panels/js/sidebar.jsx]: nit: `let` and just `sdkDriver` without `: sdkDriver`

File: add-on/panels/test/sidebar_test.js [hunk 4]
```diff
@@ -85,6 +85,10 @@ describe("loop.sidebar", function() {
 
     dispatcher = new loop.Dispatcher();
     sinon.stub(dispatcher, "dispatch");
+
+    remoteCursorStore = new loop.store.RemoteCursorStore(dispatcher, {
+          sdkDriver: {}
```
Comment (Mardak) [inline add-on/panels/test/sidebar_test.js]: nit: 4 spaces too many

File: add-on/panels/test/sidebar_test.js [hunk 5]
```diff
@@ -203,6 +215,52 @@ describe("loop.sidebar", function() {
...
+    });
+
+    it("should request ClickRemoteCursor when click event detected", function() {
+      mountTestComponent();
+
+      remoteCursorStore.setStoreState({
+        "remoteCursorClick": true
+      });
+
+      sinon.assert.calledOnce(clickRemoteCursorStub);
```
Comment (Mardak) [inline add-on/panels/test/sidebar_test.js:252]: should we additionally check that the click is set to `false` ?

Comment (Standard8) [inline add-on/panels/test/sidebar_test.js:252]: yes, added.

**Pull request merged**